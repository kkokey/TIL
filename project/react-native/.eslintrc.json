{
  "env": {
    "browser": true,
    "commonjs": true,
    "es6": true,
    "node": true,
    "jest": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:react/recommended"
  ],
  "parser": "babel-eslint",
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "ecmaVersion": 2018,
    "sourceType": "module"
  },
  "plugins": [
    "react"
  ],
  "rules": {
    "no-console": "warn",
    "indent": [
      "error",
      2
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "semi": [
      "error",
      "always"
    ],
    "react/jsx-uses-react": "error",
    "react/jsx-uses-vars": "error",
    "react/prop-types": [
      0
    ]
  }
}
