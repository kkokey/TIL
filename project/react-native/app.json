{
  "expo": {
    "name": "react-native",
    "slug": "react-native",
    "privacy": "public",
    "sdkVersion": "31.0.0",
    "ios": {
      "bundleIdentifier": "com.reactNative.sample",
      "supportsTablet": true
    },
    "android": {
      "package": "com.reactNative.sample"
    },
    "platforms": [
      "ios",
      "android"
    ],
    "version": "0.4.0",
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "description": ""
  }
}