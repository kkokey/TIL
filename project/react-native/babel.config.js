module.exports = function(api) {
  api.cache(true);
  return {
    presets: [
      ['module:metro-react-native-babel-preset'],
      'babel-preset-expo'
    ],
    plugins: [
      '@babel/plugin-transform-runtime',
      ['@babel/plugin-transform-flow-strip-types'],
      ['@babel/plugin-proposal-decorators', { legacy: true }],
      ['@babel/plugin-proposal-class-properties', { loose: true }]
    ],
  };
};
