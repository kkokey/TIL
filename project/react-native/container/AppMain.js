import React from 'react';
import {
  ScrollView,
  StyleSheet,
  SafeAreaView,
  View,
  Text,
} from 'react-native';
import ActionButton from 'react-native-action-button';
import WalletList from '../component/List';
import WalletsService from '../stores/service/ListEx'
import Modal from 'react-native-modal';
import { inject } from 'mobx-react';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f2f2f2',
  },
  main: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  btnAlign: {
    flexDirection: 'row',
  }
});

@inject('ListStore')
class AppMain extends React.Component {
  static navigationOptions = () => ({
    title: 'Sample',
  });

  constructor(props) {
    super(props);
    this.state = {
      wallets: [],
      isModalVisible: false
    };
  }

  render() {
    // const navigation = this.props.navigation;
    return (
      <SafeAreaView style={styles.container}>
        <ScrollView style={styles.container}>
          <View>
            <Text>MAIN</Text>
          </View>
        </ScrollView>
      </SafeAreaView>
    );
  }
}

export default AppMain;
