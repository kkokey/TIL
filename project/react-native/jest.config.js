// jest.config.js
module.exports = {
  preset: "react-native",
  transform: {
    "^.+\\.jsx?$": "<rootDir>/node_modules/babel-jest"
  },
  transformIgnorePatterns: [
    "/node_modules/(?!react-native)/.+"
  ],
  testRegex: "(/__tests__/.*|(\\.|/)(test|spec))\\.(jsx?|tsx?)$",
  moduleFileExtensions: ["ts", "tsx", "js", "jsx", "json", "node"]
};
