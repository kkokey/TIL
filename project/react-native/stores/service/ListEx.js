import { Utils } from '../../constants/Utils';
import axios from 'axios';
import Wallet from '../model/modelEx';

class ListEx {
  constructor() {
    this.state = {
      wallets: []
    };
    /* new ListModel() ... */
  }

  async ExListPost() {
    let postData = 'enddate: "20181212103125"\nfrom: 0\nsearchText: ""\nsearchType: "all"\nstartdate: "20181115"\ntype: "t15"';
    return await new Promise((resolve, reject) => {
      axios.post('https://globalmonitor.einfomax.co.kr/apis/usa/news/getnewsall', postData)
        .then(res => {
          resolve(res.data.data);
        }).catch(err => {
        reject(err);
      });
    });
  }

  async ExListGet(){
    return await new Promise((resolve, reject) => {
      axios.get('http://188.166.232.232:8080/weatherlist')
        .then(res => {
          console.log(res.data);

          resolve(res.data);
        }).catch(error => {
          reject(error)
        });
    });
  }
}

export default ListEx;
