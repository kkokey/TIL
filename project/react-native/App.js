import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { createAppContainer } from "react-navigation";
import { Provider } from 'mobx-react';
import AppNavigator from './navigation/Navigator';
import stores from "./stores";

const AppContainer = createAppContainer(AppNavigator);

export default class App extends React.Component {
  render() {
    return (
      <Provider {...stores}>
        <AppContainer
          ref={nav => {
            this.navigator = nav;
          }}
        />
      </Provider>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
