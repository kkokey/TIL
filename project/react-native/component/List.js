import React from 'react';
import { action } from 'mobx';
import { observer } from 'mobx-react';
import { FlatList, Text, StyleSheet, View, Image } from 'react-native';
import Touchable from 'react-native-platform-touchable';
import { Utils } from '../constants/Utils';

const walletListStyles = StyleSheet.create({
  container: {
    flex: 1,
    paddingTop: 15,
  },
  optionsTitleText: {
    fontSize: 10,
    marginLeft: 15,
    marginTop: 9,
    marginBottom: 12,
  },
  optionIconContainer: {
    marginRight: 9,
  },
  option: {
    backgroundColor: '#fdfdfd',
    paddingHorizontal: 15,
    paddingVertical: 15,
    borderBottomWidth: StyleSheet.hairlineWidth,
    borderBottomColor: '#EDEDED',
  },
  optionText: {
    fontSize: 13,
    margin: 3,
  },
  addressText: {
    fontSize: 9,
    margin: 3,
  },
});

@observer class List extends React.Component {
  // This method will be called whenever a new object is created.
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      refreshing: false,
      data: this.props.listStore
    };
    const promiseList = this.props.listStore.ExListGet();
    promiseList.then(val => {
      const rsJson = JSON.parse(val);
      this.setState({
        data: rsJson.data
      });
    });
  }

  // static getDerivedStateFromProps(nextProps, prevState) {
  //   return {
  //     data: prevState.data
  //   };
  // }


  // componentDidMount() {
  //   this.fetchFromApi('', '', '');
  // }

  // getSnapshotBeforeUpdate(prevProps, prevState) {
  //   console.log('getSnapshotBeforeUpdate prevProps');
  //   console.log('getSnapshotBeforeUpdate prevState');
  // }

  // componentDidUpdate(prevProps, prevState, snapshot) {
  //   console.log('componentDidUpdate prevState');
  //   console.log('>componentDidUpdate');
  // }

  // shouldComponentUpdate(nextProps, nextState) {
  // }

  // componentDidCatch(error, info) {
  //   // fallback UI
  //   console.error(error);
  //   console.info(info);
  // }


  // setState(state, callback) {
  //   console.log('=======================================state');
  //   console.log('state=======================================');
  //   super.setState(state, callback);
  // }

  // componentWillUnmount() {
  //   clearInterval(this.interval);
  // }

  @action
  async fetchHistoryFromApi(accountAddr){
    return '';
  }

  @action
  async fetchFromApi(addr, pubKey, priKey) {
    let resPromise = this.props.listStore.ExListPost();
    await resPromise.then( (promiseData) => {

    });
  }

  @action
  handlePressAccount() {
    this.props.navigation.navigate('main', { token: true });
  }

  render() {
    const { navigate } = this.props.navigation;
    return (
      <FlatList
        data={this.state.data}
        renderItem={({ item }) =>
          <Touchable
            style={walletListStyles.option}
            background={Touchable.Ripple('#ccc', false)}
            onPress={() => {
              navigate('detail', { selectItemName: item })
            }}>
            <View style={{ flexDirection: 'row' }}>
              <View style={{ flexDirection: 'row' }}>
                <View style={walletListStyles.optionIconContainer}>
                  <Text>{item}</Text>
                </View>
              </View>
            </View>
          </Touchable>
        }
        keyExtractor={(item, index) => index.toString()}
        extraData={this.state}
      />
    );
  }

  @action
  handleFormSubmit = e => {
    console.info(e)
  };
}

export default List;
