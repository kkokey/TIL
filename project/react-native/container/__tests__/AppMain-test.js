import 'react-native';
import React from 'react';
import AppMain from '../AppMain';
import renderer from 'react-test-renderer';
import ListEx from "../../stores/service/ListEx";
import List from "../../component/List";

describe('App screen', () => {
  it('should renders AppMain', () => {
    const store = new ListEx();
    const navigation = { navigate: jest.fn() };
    const mainRenderJSON = renderer.create(<AppMain listStore={store} navigation={navigation}/>).toJSON();

    expect(mainRenderJSON).toMatchSnapshot();
  });
});
