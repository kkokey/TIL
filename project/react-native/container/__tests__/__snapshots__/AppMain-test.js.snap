// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App screen should renders AppMain 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "backgroundColor": "#f2f2f2",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "#f2f2f2",
        "flex": 1,
      }
    }
  >
    <View>
      <View>
        <Text>
          MAIN
        </Text>
      </View>
      <RCTScrollView
        disableVirtualization={false}
        extraData={
          Object {
            "data": undefined,
            "name": "",
            "refreshing": false,
          }
        }
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={false}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        numColumns={1}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
